### ISSUE TITLE
Windows bugs - most fixed

### ISSUE THREAD

**Issue by loopy750** [overdodactyl/ShadowBird#3]
Thought I'd try the approach of leaving all your code intact and adding a "patch". Feel free to check out my file **patch_windows.css** and use that. I've also removed all the Windows-only (Firefox?) code from **context_menus.css** as that's not needed. It fixes most display bugs for Windows 10 and should keep most people happy, for now.

**Comment by overdodactyl** [comment]
Thanks @loopy750, really appreciate the help!

I just merged your changes into the repo

**Comment by loopy750** [comment]
Cool. Also, my bad for changing user.js to prefs.js in README.md, I never knew about that one until now.

**Comment by overdodactyl** [comment]
No worries! Just reverted back to `user.js`; I added an extra sentence clarifying that users might have to create the file since it doesn't exist by default.

**Comment by loopy750** [comment]
Just noticed a bit of a problem with userchrome.css, it needs the windows fixes right at the bottom of the file so they can override any previous code. What's the easiest way to make sure that always happens? Is there a gulp task that can do that? Also I'll have an updated patch_windows soon that should fix just about all the remaining bugs.

**Comment by overdodactyl** [comment]
Ah, good catch! I've just altered the gulp task `publish` so that it should always be appended last

Edit: The task `userChrome` is actually what was edited, but that is called by `publish`

**Comment by loopy750** [comment]
Completed all the fixes needed for Windows that i could find. In future is there any sort of "official" method to get your attention when I have something new? Although I imagine it will be nowhere near as frequent as Firefox

**Comment by OdinGitDat** [comment]
thanks loopy, works great!

**Comment by overdodactyl** [comment]
Just merged the latest version; thanks again, it's much appreciated!

In terms of changes in the future, I'm good with whatever is easiest for you.  I have notifications turned on for this repo, so I should get notified pretty quickly if there are any pull requests or issues opened.

I could also just add you onto the project as a collaborator with write access to the repo so you can make changes directly on here.

Alternatively, there's my email address, reddit account or the slack group that was set up for ShadowFox but could be repurposed for this and that.

Just let me know whatever is best for you!

**Comment by loopy750** [comment]
I think pull requests sounds like the way to go, you'll just have to explain to me exactly how that works. I'll try and keep up-to-date with yours, either way chances are you'll just need patch_windows.css

**Comment by overdodactyl** [comment]
These two short articles by GitHub probably do a better job at explaining PRs than I could:

- [About pull requests](https://help.github.com/articles/about-pull-requests/)
- [Creating a pull request](https://help.github.com/articles/creating-a-pull-request/)

PRs are based on an entire branch though, not specific files or commits; in other words if a PR is sent from your master, it would include all changes you've made to the repo.  If you make changes you don't think would want to be included here, you could either use a separate branch and submit the PR from that one, or you could send the PR from master, and I can use [git-cherry-pick](https://git-scm.com/docs/git-cherry-pick) to pick out the relavent commits.  If a single commit includes both changes that should be applied to this repo and ones that shouldn't, I think I should be able to use [rebasing](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) to separate things out. 

In short, if you don't want to worry about branching and all that, or if your repo isn't up to date with this one, just submit a PR from your master branch and leave a comment in the PR mentioning what changes I should try and include.  Generally speaking though, it will be easiest to merge if single commits don't include both changes that should be included here and ones that shouldn't.
